dev/sdk_go.go

```diff
@@ -19,21 +20,49 @@ type GoSDK struct {
 }
 
 // Lint the Go SDK
-func (t GoSDK) Lint(ctx context.Context) error {
+func (t GoSDK) Lint(ctx context.Context) (rerr error) {
+	ctx, span := Tracer().Start(ctx, "lint the Go SDK")
+	defer func() {
+		if rerr != nil {
+			span.SetStatus(codes.Error, rerr.Error())
+		}
+		span.End()
+	}()
 	eg, ctx := errgroup.WithContext(ctx)
-	eg.Go(func() error {
+	eg.Go(func() (rerr error) {
+		ctx, span := Tracer().Start(ctx, "lint the go source")
+		defer func() {
+			if rerr != nil {
+				span.SetStatus(codes.Error, rerr.Error())
+			}
+			span.End()
+		}()
 		return dag.
 			Go(t.Dagger.Source()).
 			Lint(ctx, dagger.GoLintOpts{Packages: []string{"sdk/go"}})
 	})
-	eg.Go(func() error {
+	eg.Go(func() (rerr error) {
+		ctx, span := Tracer().Start(ctx, "check that the generated client library is up-to-date")
+		defer func() {
+			if rerr != nil {
+				span.SetStatus(codes.Error, rerr.Error())
+			}
+			span.End()
+		}()
 		return util.DiffDirectoryF(ctx, t.Dagger.Source(), t.Generate, "sdk/go")
 	})
 	return eg.Wait()
 }
 
 // Test the Go SDK
-func (t GoSDK) Test(ctx context.Context) error {
+func (t GoSDK) Test(ctx context.Context) (rerr error) {
+	ctx, span := Tracer().Start(ctx, "test the Go SDK")
+	defer func() {
+		if rerr != nil {
+			span.SetStatus(codes.Error, rerr.Error())
+		}
+		span.End()
+	}()
```

Comment:
Personally think this is a step too far, since this span is 1:1 with the original function call, and will add an extra step to expand while jumping into its contents[^1]. I wouldn't expect every user to do with this with every function, so I don't think we should either.

Agree that we should be using function descriptions somehow in Cloud, but don't think they're a total substitue for the function call either, since the args may be significant. Maybe we could always show the description once expanded and/or in a tooltip?

[^1]: Granted you'd probably be just jumping straight to the error in practice, so maybe that doesn't matter.

Reply:
Yeah I agree

Reply:
> Agree that we should be using function descriptions somehow in Cloud, but don't think they're a total substitue for the function call either, since the args may be significant.

I think a toggle would be amazing. Toggle between "text view" (only show text descriptions, either from custom spans or scraped from comments) and "function view" (function calls only, no custom spans). Implemented in both TUI and web. I think it could be magical.